lib/ansible/plugins/inventory/gcp_compute.py

```diff
@@ -0,0 +1,169 @@
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: gcp_compute
+    plugin_type: inventory
+    short_description: Google Cloud Compute Engine inventory source
+    extends_documentation_fragment:
+        - gcp
+    description:
+        - Get inventory hosts from Google Cloud Platform GCE.
+        - Uses a <name>.gcp.yaml (or <name>.gcp.yml) YAML configuration file.
+    options:
+        zones:
+          description: A list of regions in which to describe GCE instances.
+        filters:
+          description: A dictionary of filter value pairs. Available filters are listed here
+              U(https://cloud.google.com/compute/docs/reference/rest/v1/instances/list)
+'''
+
+EXAMPLES = '''
+simple_config_file:
+
+    plugin: gcp_compute
+    zones: # populate inventory with instances in these regions
+      - us-east1-a
+    filters:
+      machine-type:
+        eq:
+          - n1-standard-1
+'''
+
+from ansible.errors import AnsibleError, AnsibleParserError
+from ansible.module_utils._text import to_native, to_text
+from ansible.module_utils.six import string_types
+from ansible.module_utils.gcp_utils import GcpSession, navigate_hash, GcpRequestException
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable, to_safe_group_name
+import json
+
+
+# The mappings give an array of keys to get from the filter name to the value
+# returned by boto3's GCE describe_instances method.
+class GcpMockModule(object):
+    def __init__(self, params):
+        self.params = params
+
+    def fail_json(self, *args, **kwargs):
+        raise AnsibleError(kwargs['msg'])
+
+
+class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable):
```

Comment:
Constructable and Cacheable features are not implemented yet.

After calling `self._read_config_data` you can access Cacheable and Constructed options with self.get_option(option_name). If the user hasn't provided those options get_option will return the default (you can see those in the doc fragments: https://github.com/ansible/ansible/blob/devel/lib/ansible/utils/module_docs_fragments/constructed.py, https://github.com/ansible/ansible/blob/devel/lib/ansible/utils/module_docs_fragments/inventory_cache.py. 

You'll need to add constructed and inventory_cache under `extends_documentation_fragment` to be able to use them like that.

Reply:
I added Cache support. I'm not entirely sure how Constructed works after looking through the AWS module.